eth/state_accessor.go

```diff
@@ -172,8 +172,9 @@ func (eth *Ethereum) hashState(ctx context.Context, block *types.Block, reexec u
 		if err != nil {
 			return nil, nil, fmt.Errorf("state reset after block %d failed: %v", current.NumberU64(), err)
 		}
-		// Note: In subnet-evm, the state reference is held by passing true to [statedb.Commit].
-		// Drop the parent state to prevent accumulating too many nodes in memory.
+		// Hold the state reference and also drop the parent state
+		// to prevent accumulating too many nodes in memory.
+		tdb.Reference(root, common.Hash{})
```

Comment:
I reverted the logic to having a separate call to Reference here instead of using a triedb with ref. counting.
This is because the only triedb that is used in this fn (hashState) is either created here or passed in from return value of this fn.
Basically the live chain db is not used to re-execute blocks, per my understanding